content/zh/blog/2023/egress-sni/index.md

```diff
@@ -0,0 +1,455 @@
+但由于我们在测试环境中没有这样做，所以让我们访问另一个外部站点进行比较：
+
+{{< text bash >}}
+$ kubectl exec "$SOURCE_POD" -c sleep -- sh -c 'curl -s https://cloud.ibm.com/login | grep -o "<title>.*</title>"'
+<title>IBM Cloud</title>
+{{< /text >}}
+
+由于我们在全局范围内打开了访问日志记录（使用清单中的 `Telemetry` CR），
+因此我们现在可以检查日志以了解代理如何处理上述请求。
+
+首先，检查网关日志：
+
+{{< text bash >}}
+$ kubectl logs -n istio-egress $GATEWAY_POD
+[...]
+[2023-11-24T13:21:52.798Z] "- - -" 0 - - - "-" 813 111152 55 - "-" "-" "-" "-" "185.15.59.224:443" dynamic_forward_proxy_cluster 172.17.5.170:48262 envoy://sni_listener/ envoy://internal_client_address/ en.wikipedia.org -
+[2023-11-24T13:21:52.798Z] "- - -" 0 - - - "-" 1531 111950 55 - "-" "-" "-" "-" "envoy://sni_listener/" sni_cluster envoy://internal_client_address/ 172.17.5.170:8443 172.17.34.35:55102 outbound_.443_.wildcard_.egressgateway.istio-egress.svc.cluster.local -
+[2023-11-24T13:21:53.000Z] "- - -" 0 - - - "-" 821 92848 49 - "-" "-" "-" "-" "185.15.59.224:443" dynamic_forward_proxy_cluster 172.17.5.170:48278 envoy://sni_listener/ envoy://internal_client_address/ de.wikipedia.org -
+[2023-11-24T13:21:53.000Z] "- - -" 0 - - - "-" 1539 93646 50 - "-" "-" "-" "-" "envoy://sni_listener/" sni_cluster envoy://internal_client_address/ 172.17.5.170:8443 172.17.34.35:55108 outbound_.443_.wildcard_.egressgateway.istio-egress.svc.cluster.local -
+{{< /text >}}
+
+这里有四条日志，代表上面三个 curl 请求中的两个。每对日志都显示单个请求如何流经 Envoy 流量处理管道。
+它们以相反的顺序被打印，但我们可以看到第 2 行和第 4 行显示请求到达网关服务并通过内部 `sni_cluster` 目标传递。
+第 1 行和第 3 行显示最终目标是根据内部 SNI 标头确定的，即应用程序设置的目标主机。
+请求被转发到 `dynamic_forward_proxy_cluster`，后者最终将请求从 Envoy 发送到远程目标。
+
+很好，但是对 IBM Cloud 的第三个请求在哪里？让我们检查一下 Sidecar 日志：
+
+{{< text bash >}}
+$ kubectl logs $SOURCE_POD -c istio-proxy
+[...]
+[2023-11-24T13:21:52.793Z] "- - -" 0 - - - "-" 813 111152 61 - "-" "-" "-" "-" "172.17.5.170:8443" outbound|443|wildcard|egressgateway.istio-egress.svc.cluster.local 172.17.34.35:55102 208.80.153.224:443 172.17.34.35:37020 en.wikipedia.org -
+[2023-11-24T13:21:52.994Z] "- - -" 0 - - - "-" 821 92848 55 - "-" "-" "-" "-" "172.17.5.170:8443" outbound|443|wildcard|egressgateway.istio-egress.svc.cluster.local 172.17.34.35:55108 208.80.153.224:443 172.17.34.35:37030 de.wikipedia.org -
+[2023-11-24T13:21:55.197Z] "- - -" 0 - - - "-" 805 15199 158 - "-" "-" "-" "-" "104.102.54.251:443" PassthroughCluster 172.17.34.35:45584 104.102.54.251:443 172.17.34.35:45582 cloud.ibm.com -
+{{< /text >}}
+
+正如您所看到的，到 Wikipedia 的请求是通过网关发送的，
+而到 IBM Cloud 的请求是直接从应用程序 Pod 发送到互联网的，如 `PassthroughCluster` 日志所示。
+
+## 总结 {#conclusion}
+
+我们使用出口网关实现了出口 HTTPS/TLS 流量的受控路由，支持任意域名和通配符域名。
+在生产环境中，本文中展示的示例将进行扩展以支持 HA 要求
+（例如，为网关 `Deployment` 添加区域感知等）并限制应用程序的直接外部网络访问，
+以便应用程序只能通过网关访问公共网络，该网关仅限于访问一组预定义的远程主机名。
+
+该解决方案可以轻松进行扩展。您可以在配置中包含多个域名，一旦执行，它们就会被列入白名单！
+无需配置每个域的 `VirtualService` 或其他路由详细信息。但要小心的是，
+由于域名在配置中的多个位置被列出。如果您使用 CI/CD 工具（例如 Kustomize），
+最好将域名列表提取到一个单一位置，使其被渲染到所需的配置资源中。
+
+就是这些！我希望这可以帮到您。如果您是之前基于 Nginx 解决方案的现有用户，
+现在可以在升级到 Istio 1.20 之前迁移到此方法，否则您当前的设置会被破坏。
+
+请开心的使用 SNI 路由！
+
+## References
```

Comment:
> Need help:
> 
> ```
> Further documentation is available for these failures:
>  - MD022: https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md022---headers-should-be-surrounded-by-blank-lines
> ./zh/blog/2023/egress-sni/index.md:14:1.13 版的[官方 Istio 文档](https://istio.io/v1.13/zh/docs/tasks/traffic-management/egress/wildcard-egress-hosts/#wildcard-configuration-for-arbitrary-domains)中，
> ```
> 
> How to solve this lint problem.

Is it related to this extra line? Not sure though.

Reply:
Sorry, the full message is:
```
Further documentation is available for these failures:
 - MD022: https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md022---headers-should-be-surrounded-by-blank-lines
./zh/blog/2023/egress-sni/index.md:14:1.13 版的[官方 Istio 文档](https://istio.io/v1.13/zh/docs/tasks/traffic-management/egress/wildcard-egress-hosts/#wildcard-configuration-for-arbitrary-domains)中，
**Ensure markdown content uses relative references to istio.io**
Running ["HtmlCheck", "ImageCheck", "OpenGraphCheck", "ScriptCheck"] on ["./public"] on *.html... 
```